packages/core/src/diff/changes/directive-usage.ts

```diff
@@ -0,0 +1,206 @@
+    },
+    type: ChangeType.DirectiveUsageUnionMemberAdded,
+    message: `Directive '${args.meta.addedUnionMemberTypeName}' was added to Union type '${args.meta.unionName}'`,
+    meta: args.meta,
+    path: args.meta.unionName,
+  } as const;
+}
+
+export function directiveUsageUnionMemberAdded(
+  union: GraphQLUnionType,
+  directive: ConstDirectiveNode,
+): Change<ChangeType.DirectiveUsageUnionMemberAdded> {
+  return directiveUsageBuildUnionMemberAddedMessageFromMeta({
+    type: ChangeType.DirectiveUsageUnionMemberAdded,
+    meta: {
+      unionName: union.name,
+      addedUnionMemberTypeName: directive.name.value,
+    },
+  });
+}
+
+export function directiveUsageBuildUnionMemberRemovedMessageFromMeta(
+  args: DirectiveUsageUnionMemberRemovedChange,
+) {
+  return {
+    criticality: {
+      level: CriticalityLevel.Dangerous,
+      reason:
+        'Removing a union member from a union can cause existing queries that use this union member in a fragment spread to error.',
+    },
+    type: ChangeType.DirectiveUsageUnionMemberRemoved,
+    message: `Directive '${args.meta.removedUnionMemberTypeName}' was removed from Union type '${args.meta.unionName}'`,
+    meta: args.meta,
+    path: args.meta.unionName,
+  } as const;
+}
+
+export function directiveUsageUnionMemberRemoved(
+  union: GraphQLUnionType,
+  directive: ConstDirectiveNode,
+): Change<ChangeType.DirectiveUsageUnionMemberRemoved> {
+  return directiveUsageBuildUnionMemberRemovedMessageFromMeta({
+    type: ChangeType.DirectiveUsageUnionMemberRemoved,
+    meta: {
+      unionName: union.name,
+      removedUnionMemberTypeName: directive.name.value,
+    },
+  });
+}
+
+export function directiveUsageEnumAddedFromMeta(args: DirectiveUsageEnumAddedChange) {
+  return {
+    type: ChangeType.DirectiveUsageEnumAdded,
+    criticality: {
+      level: CriticalityLevel.Dangerous,
+      reason: `Directive '${args.meta.addedDirectiveName}' was added to Enum '${args.meta.enumName}'`,
+    },
+    message: `Directive '${args.meta.addedDirectiveName}' was added to Enum '${args.meta.enumName}'`,
+    meta: args.meta,
+    path: [args.meta.enumName, args.meta.addedDirectiveName].join('.'),
+  } as const;
+}
+
+export function directiveUsageEnumAdded(
+  newEnum: GraphQLEnumType,
+  directive: ConstDirectiveNode,
+): Change<ChangeType.DirectiveUsageEnumAdded> {
+  return directiveUsageEnumAddedFromMeta({
+    type: ChangeType.DirectiveUsageEnumAdded,
+    meta: {
+      enumName: newEnum.name,
+      addedDirectiveName: directive.name.value,
+    },
+  });
+}
+
+export function directiveUsageEnumRemovedFromMeta(args: DirectiveUsageEnumRemovedChange) {
+  return {
+    type: ChangeType.DirectiveUsageEnumRemoved,
+    criticality: {
+      level: CriticalityLevel.Dangerous,
+      reason: `Directive '${args.meta.removedDirectiveName}' was removed from Enum '${args.meta.enumName}'`,
```

Comment:
Reason should not be the same as the message, right?
Also, Enum -> enum (same for all other keywords)